inc/connector/namespace.php

```diff
@@ -382,16 +382,39 @@ function pre_get_scheduled_event( $pre, $hook, $args, $timestamp ) {
 	}
 
 	$jobs = Job::get_jobs_by_query( [
+		'args' => null,
+		'hook' => null,
+		'limit' => 100,
```

Comment:
What happens when there's 100 jobs in the table with hook `foo` and we're looking for the hook `bar` that happens to be at position 101?

Reply:
I was wondering the same but it's a problem we'd see in other situations. I guess that number was chosen because by default we're only looking for the `waiting` and `running` statuses.

See also:

https://github.com/humanmade/Cavalcade/blob/master/inc/connector/namespace.php#L294-L306

https://github.com/humanmade/Cavalcade/blob/master/inc/connector/namespace.php#L420-L430

Reply:
@rmccue able to comment on this? There is at least a filter so if a site has more than 100 combinations of hooks & args that are upcoming or running you could filter the SQL `jobs_by_query` args.